pkg/apis/core/validation/validation_test.go

```diff
@@ -7146,23 +7146,21 @@ func TestValidateVolumeMounts(t *testing.T) {
 	}
 
 	errorCases := map[string][]core.VolumeMount{
-		"empty name":                                       {{Name: "", MountPath: "/foo"}},
-		"name not found":                                   {{Name: "", MountPath: "/foo"}},
-		"empty mountpath":                                  {{Name: "abc", MountPath: ""}},
-		"mountpath collision":                              {{Name: "foo", MountPath: "/path/a"}, {Name: "bar", MountPath: "/path/a"}},
-		"absolute subpath":                                 {{Name: "abc", MountPath: "/bar", SubPath: "/baz"}},
-		"subpath in ..":                                    {{Name: "abc", MountPath: "/bar", SubPath: "../baz"}},
-		"subpath contains ..":                              {{Name: "abc", MountPath: "/bar", SubPath: "baz/../bat"}},
-		"subpath ends in ..":                               {{Name: "abc", MountPath: "/bar", SubPath: "./.."}},
-		"disabled MountPropagation feature gate":           {{Name: "abc", MountPath: "/bar", MountPropagation: &propagation}},
-		"name exists in volumeDevice":                      {{Name: "xyz", MountPath: "/bar"}},
-		"mountpath exists in volumeDevice":                 {{Name: "uvw", MountPath: "/mnt/exists"}},
-		"both exist in volumeDevice":                       {{Name: "xyz", MountPath: "/mnt/exists"}},
-		"rro but not ro":                                   {{Name: "123", MountPath: "/rro-bad1", ReadOnly: false, RecursiveReadOnly: ptr.To(core.RecursiveReadOnlyEnabled)}},
-		"rro with incompatible propagation":                {{Name: "123", MountPath: "/rro-bad2", ReadOnly: true, RecursiveReadOnly: ptr.To(core.RecursiveReadOnlyEnabled), MountPropagation: ptr.To(core.MountPropagationHostToContainer)}},
-		"rro-if-possible but not ro":                       {{Name: "123", MountPath: "/rro-bad1", ReadOnly: false, RecursiveReadOnly: ptr.To(core.RecursiveReadOnlyIfPossible)}},
-		"subPath not allowed for image volume sources":     {{Name: "image-volume", MountPath: "/image-volume-err-1", SubPath: "/foo"}},
-		"subPathExpr not allowed for image volume sources": {{Name: "image-volume", MountPath: "/image-volume-err-2", SubPathExpr: "$(POD_NAME)"}},
```

Comment:
possible to add new success cases for these?

Reply:
Yes, added 2 succeeding and one failure test case.